### PR TITLE
Automate the installation of nightly rustfmt in CI

### PR DESCRIPTION
**Pull request by raskchanky**
Closes https://github.com/habitat-sh/habitat/issues/6170
Signed-off-by: Josh Black <raskchanky@gmail.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

**Comment by raskchanky** [comment]
@baumanj I was thinking about this last night.  One thing I could do to mitigate the infinite loop issue is to only try X days in the past, e.g. 90.  It seems highly unlikely that nightly rust would go 90 days without a functioning `rustfmt`.

File: support/ci/rustfmt.sh [hunk 1]
```diff
@@ -8,16 +8,38 @@ export PATH="/opt/rust/bin:$PATH"
...
+
+  if sudo -E $rustup component add --toolchain "$toolchain" rustfmt; then
+    echo "Rust $toolchain has rustfmt. Excellent!"
+    break
+  else
+    date_count=$((date_count + 1))
+    echo "Rust $toolchain did not include rustfmt. Let's try $date_count day(s) ago."
+  fi
+
+  if [ "$date_count" -eq "$max_days" ]; then
```
[the remark this answers is not in the document]
Comment (raskchanky) [reply support/ci/rustfmt.sh]: Gotcha, makes sense.  I'll open a separate PR for this change.  Thanks for the suggestions!

[the remark this answers is not in the document]
Comment (raskchanky) [reply support/ci/rustfmt.sh]: https://github.com/habitat-sh/habitat/pull/6193

**Pull request merged**